ambari-server/src/main/java/org/apache/ambari/server/state/host/HostImpl.java

```diff
@@ -1256,14 +1256,14 @@ public void calculateHostStatus(Long clusterId) throws AmbariException {
       String category = componentInfo.getCategory();
```

Comment:
Should there be some sort of logic to handle the case where `category` is `null`.... or is ignoring such components in this calculation ok?

Reply:
Is logging as an error or warning would be enough?

Reply:
A logged message would be nice.  However do we know what the effect is if a component is not accounted for due to a missing category.  Maybe this is a real show stopper and some exception should be thrown.  

@mpapirkovskyy can you comment on this?

Reply:
Added logging when category is null. Please review @rlevas

Reply:
@rlevas I'm not sure if I can help here.
I don't even get what null category means, since it looks like we never had such components before.

@jonathan-hurley Since Java 7 theres Objects.equals(), you no longer need external lib for safe and easy check.

Reply:
I guess null category means faulty service definition, which should be fixed, but the goal is to handle it more gracefully.

Reply:
Oh sure - Objects.equals() is fine too - I'm just used to StringUtils :)